pkg/security/apiserver/securitycontextconstraints/matcher.go

```diff
@@ -36,28 +36,40 @@ func NewDefaultSCCMatcher(c securitylisters.SecurityContextConstraintsLister, au
 }
 
 // FindApplicableSCCs implements SCCMatcher interface
-func (d *defaultSCCMatcher) FindApplicableSCCs(userInfo user.Info, namespace string) ([]*securityapi.SecurityContextConstraints, error) {
+// It finds all SCCs that the subjects in the `users` argument may use.
+// The returned SCCs are sorted by priority.
+func (d *defaultSCCMatcher) FindApplicableSCCs(namespace string, users ...user.Info) ([]*securityapi.SecurityContextConstraints, error) {
 	var matchedConstraints []*securityapi.SecurityContextConstraints
 	constraints, err := d.cache.List(labels.Everything())
 	if err != nil {
 		return nil, err
 	}
-	for _, constraint := range constraints {
-		if constraintAppliesTo(constraint, userInfo, namespace, d.authorizer) {
-			matchedConstraints = append(matchedConstraints, constraint)
+
+	// filter out SCCs if we got some users, leave as is if not
+	if users == nil {
```

Comment:
You'll need `users == nil || len(users) < 1` to catch cases where it's called with something like `FindApplicableSCCs(ns, users...)` and `users` is empty.

Reply:
I'm going to replace the condition with just `len(users) == 0`, `len()` should be handling nil internally.